tests/wasm/coverage/report.ts

```diff
@@ -0,0 +1,59 @@
+import fs from 'fs'
+import md5File from 'md5-file'
+import { InstrumentedLines } from './'
+
+const COVERAGE_FOLDER = './coverage/'
+const COVERAGE_FILE = COVERAGE_FOLDER + 'report.json'
+
+interface ReportData {
+  repo_token: string | undefined
+  service_job_id: string | undefined
+  service_name: string
+  source_files: {
+    name: string
+    source_digest: string
+    coverage: (number | null)[]
+  }[]
+}
+
+export class Report {
+  data: ReportData
+
+  constructor(data?: ReportData) {
+    if (!data) {
+      data = {
+        repo_token: process.env.COVERALLS_REPO_TOKEN,
+        service_job_id: process.env.TRAVIS_BUILD_ID,
+        service_name: 'travis-ci',
+        source_files: [],
+      }
+    }
+    this.data = data
+  }
+
+  addFile(file: string, instrumentedLines: InstrumentedLines): void {
+    const source_digest = md5File.sync(file)
+    this.data.source_files.push({
+      name: file,
+      source_digest,
+      coverage: instrumentedLines.map((isInstrumented: boolean) => (isInstrumented ? 0 : null)),
+    })
+  }
+
+  write(): void {
+    fs.mkdirSync(COVERAGE_FOLDER, { recursive: true })
```

Comment:
Maybe add a note as to why this method can't be `async`? I imagine its because it needs to be called from Wasm in a way that isn't `Promise`-aware?